torch/csrc/distributed/autograd/context/thread_local_context.h

```diff
@@ -0,0 +1,23 @@
+#pragma once
+
+#include <torch/csrc/distributed/autograd/context/context.h>
+
+namespace torch {
+namespace distributed {
+namespace autograd {
+
+class ThreadLocalDistAutogradContext {
```

Comment:
Add some docs explaining the purpose of this class and how it should be used.

Reply:
Will do.